cpp/ScanChain_Zp.C

```diff
@@ -1055,6 +1062,16 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process, in
           double muonResSmearFactor = rnd_muonMomRes.Gaus(0,muonResSmearParam*0.46);
           Muon_p4[mu].SetPt( Muon_p4[mu].Pt()*(1 + 0.5*muonResUnc*muonResSmearFactor) ); // 0.5*muonResUnc = sign of variation
         }
+        if ( isMC && muonScaleUnc != 0 ) { // 2 means that variation are to be applied
+          // https://twiki.cern.ch/twiki/bin/view/CMS/MuonUL2016#Momentum_Scale
+          // https://gitlab.cern.ch/cms-muonPOG/GeneralizedEndpoint/GEScaleSyst
+          if ( year=="2018" || year.Contains("2016") ) { } // Only available for 2017!
+          else {
```

Comment:
A "positive" if statement may be cleaner: i.e., only one `if (year=="2017") {`, removing the line above?

Reply:
Agreed, not sure what was the logic I did it like that.